### DISCUSSION TITLE
Conditional Application.Resources resource

### DISCUSSION THREAD

**Discussion by lukaszossowski** [AvaloniaUI/Avalonia#9365] [Q&A] [answered]
Hi,

I am writing a crossplatform app targeted for desktop and Android. Most of the resources deifned in the App.axaml file in Application.Resources section apply to both dekstop and mobile implementations. However there are a couple of values that I would like to split, depending on if I am running a mobile or desktop version. 

Currently what I'm doing is this:

```
  public override void OnFrameworkInitializationCompleted()
        {
            if (ApplicationLifetime is IClassicDesktopStyleApplicationLifetime desktop)
            {
                desktop.MainWindow = new MainWindow
                {
                    DataContext = new MainWindowViewModel(),
                };
            }
            else if (ApplicationLifetime is ISingleViewApplicationLifetime singleViewPlatform)
            {

                Current.Resources.Remove("HeaderWidth");
                Current.Resources.Add("HeaderWidth", 100);

               singleViewPlatform.MainView = new MainMobileView()
                {
                    DataContext = new MainWindowViewModel(),
                };
            }

            base.OnFrameworkInitializationCompleted();
        }
```

So manually removing and readding the resource with the required value. Since there is basically only one resource that I need replacing currently this is a solution, but probably not THE solution.

What is the correct way to go about this, preferably to have the resource defined twice somehow in the app.axaml file?

**Comment by maxkatz6** [comment, marked as answer]
> Since there is basically only one resource that I need replacing currently this is a solution, but probably not THE solution.

That's a solution, but not the single one, correct.
For the single resource it's not too bad, but if you have many resources, I would personally create two different resource dictionary files and add/reference them from the OnFrameworkInitializationCompleted method with the ResourceInclude from codebehind. I.e. not included by default.

But, in 11.0 (probably the next preview) we will have a OnPlatform markup extension. And I want to make it work in this way with resources too:
```xaml
<Resources>
   <OnPlatform x:Key="HeaderWidth"
               x:TypeArguments="x:Double"
               Default="200"
               Android="100" />
</Resources>
```

Which would be perfect for the single resources.

**Reply by lukaszossowski** [reply]
Thank you, then I will certainly move on to the OnPlatform extension once it's available, it is exactly what I need and I think an important step to take to make Avalonia as cross platform friendly as possible.
